static/app/components/forms/compactSelect.tsx

```diff
@@ -157,7 +164,23 @@ function CompactSelect({
   // Update the button label when the value changes
   function getLabel(newValue): React.ReactNode {
     const valueSet = Array.isArray(newValue) ? newValue : [newValue];
-    const optionSet = valueSet.map(val => options.find(opt => opt.value === val));
+
+    // Recursively finds the selected option(s) from the options list
+    function getSelectedOptions(
+      opts: Props<OptionType>['options'],
+      value: Props<OptionType>['value']
+    ): Props<OptionType>['options'] {
```

Comment:
I would try move these utility fn's outside the component (like top of the file before the props). The benefit of that is that it makes the function "static", so it's not recreated on each render, it makes it easily exportable if you wanted to test them and keeps the function body easier to read, as you dont need to focus on the utility fn's.